src/main/java/gov/nasa/pds/tools/validate/content/array/ArrayContentValidator.java

```diff
@@ -389,15 +392,17 @@ public static boolean isSpecialConstant(Number value, SpecialConstants constants
         if (value instanceof Double) {
           val = Long.valueOf(Double.doubleToRawLongBits((Double)value));
           if (con.equals (val)) {
-            return true;
+            matchEdge &= true;
```

Comment:
This is splitting a really fine hair: `&` is bitwise or boolean and `&&` is boolean. The Java Language Spec says `&` will work with both. The difference, though, is that with `&&`, there is short-circuiting. If the left side is false the right side won't even be evaluated. (Granted the right-side here is a constant.) With `&`, both sides are evaluated.

So, there's a performance consideration if the `true` is ever replaced with, say, a function call.

Anyway, no change at this time; just wanted to point it out 😇

Reply:
@nutjob4life 

Yeah, it is another disappointment with java. They just can not be bothered to overload &= for boolean too. It is funny, because short cutting is the root cause of this problem - return statements in blocks rather than just one return at the end like us dinosaurs learned a half century ago.

Reply:
Oh man @al-niessner, I sometimes long for those days 🦕 😝